DSL/conditions.md

```diff
@@ -0,0 +1,288 @@
+# Declarative DSL to Check Conditions in JSON Schema
+
+This document provides an overview of a declarative Domain Specific Language (DSL) designed to check various conditions in a JSON schema. The DSL is structured as an array of conditions, with each condition containing three mandatory fields: `path`, `operation`, and `value`.
+
+## DSL Structure
+
+### Condition Fields
+
+- **path**: Specifies the path to the target in the schema where the operation will be performed.
+- **operation**: Specifies the type of operation to perform at the target path.
+- **value**: Specifies the value to be used in the operation.
+
+## Supported Operations
+
+### Type Checks
+
+- **type-is**: Checks if the type of the target matches the specified type.
+- **contains-type**: Checks if an array contains elements of a specified type.
+
+### Equality Checks
+
+- **equals**: Checks if the target equals the specified value.
+- **not-equals**: Checks if the target does not equal the specified value.
+- **size-equals**: Check if the size of the array or number of properties of the object equal the specified value. 
+
+### Presence Checks
+
+- **has-key**: Checks if the target object has the specified key.
+
+### Array Operations
```

Comment:
Some of these sections, like this one, are type specific (i.e. array operations) while the above ones are described by their purpose (i.e. "type checks"). Let's be consistent. A way to fix it might be to group the above operations on a section for "Any" type?

Reply:
So, I will move `contains` to presence check and `min-properties` to "Any"??

Reply:
No, only the ones that apply to any instance type. For example, `min-properties` is about objects, while `type-is` applies to any instance irrespective of its type.